Assets/Scripts/Data Model/Layer.cs

```diff
@@ -1,1450 +1,1481 @@
 
-		return result;
-	}
+			if (entityCount == 0) { return new Rect(); }
 
-	public override List<Entity> GetEntitiesAt(Vector2 position)
-	{
-		List<SubEntity> subEntities = GetSubEntitiesAt(position);
-		List<Entity> result = new List<Entity>();
-		foreach (SubEntity subEntity in subEntities)
-		{
-			if (!result.Contains(subEntity.Entity))
+			Rect result = GetEntity(0).GetEntityBounds();
+			for (int i = 1; i < entityCount; ++i)
 			{
-				result.Add(subEntity.Entity);
+				Vector2 min = Vector2.Min(result.min, GetEntity(i).GetEntityBounds().min);
+				Vector2 max = Vector2.Max(result.max, GetEntity(i).GetEntityBounds().max);
+				result = new Rect(min, max - min);
 			}
+
+			return result;
 		}
-		return result;
-	}
 
-	public override SubEntity GetSubEntityByDatabaseID(int id)
-	{
-		for (int i = 0; i < GetEntityCount(); i++)
+		public override List<Entity> GetEntitiesAt(Vector2 position)
 		{
-			Entity entity = GetEntity(i);
-			for (int j = 0; j < entity.GetSubEntityCount(); j++)
+			List<SubEntity> subEntities = GetSubEntitiesAt(position);
+			List<Entity> result = new List<Entity>();
+			foreach (SubEntity subEntity in subEntities)
 			{
-				if (entity.GetSubEntity(j).GetDatabaseID() == id)
+				if (!result.Contains(subEntity.Entity))
 				{
-					return entity.GetSubEntity(j);
+					result.Add(subEntity.Entity);
 				}
 			}
+			return result;
 		}
 
-		return null;
-	}
+		public override SubEntity GetSubEntityByDatabaseID(int id)
+		{
+			for (int i = 0; i < GetEntityCount(); i++)
+			{
+				Entity entity = GetEntity(i);
+				for (int j = 0; j < entity.GetSubEntityCount(); j++)
+				{
+					if (entity.GetSubEntity(j).GetDatabaseID() == id)
+					{
+						return entity.GetSubEntity(j);
+					}
+				}
+			}
 
-	public override SubEntity GetSubEntityByPersistentID(int persistentID)
-	{
-		for (int i = 0; i < GetEntityCount(); i++)
+			return null;
+		}
+
+		public override SubEntity GetSubEntityByPersistentID(int persistentID)
 		{
-			Entity entity = GetEntity(i);
-			for (int j = 0; j < entity.GetSubEntityCount(); j++)
+			for (int i = 0; i < GetEntityCount(); i++)
 			{
-				if (entity.GetSubEntity(j).GetPersistentID() == persistentID)
+				Entity entity = GetEntity(i);
+				for (int j = 0; j < entity.GetSubEntityCount(); j++)
 				{
-					return entity.GetSubEntity(j);
+					if (entity.GetSubEntity(j).GetPersistentID() == persistentID)
+					{
+						return entity.GetSubEntity(j);
+					}
 				}
 			}
+			return null;
 		}
-		return null;
-	}
 
-	public override int GetEntityCount()
-	{
-		return Entities.Count;
-	}
+		public override int GetEntityCount()
+		{
+			return Entities.Count;
+		}
 
-	public override Entity GetEntity(int index)
-	{
-		return Entities[index];
-	}
+		public override Entity GetEntity(int index)
+		{
+			return Entities[index];
+		}
 
-	//public override int GetActiveEntityCount()
-	//{
-	//    return activeEntities.Count;
-	//}
+		//public override int GetActiveEntityCount()
```

Comment:
Commented code, as always see if it makes sense to leave it there for the future, otherwise remove it! :)